requirements.txt

```
@@ -1,3 +1,5 @@
-numpy==1.19.4; python_version >= '3.6'
-cython==0.29.21
-dimod==0.9.11
+numpy==1.19.1;python_version<="3.9"
+numpy==1.21.4;python_version>="3.10"
```

There's a gap `<3.9, 3.10>`.

---

Seems to work fine? For instance the CI 3.9.4 job correctly installs NumPy 1.19.1

---

Though I should actually change it to 3.8. Can use the other syntax if you feel like it's clearer.

---

If that works, I need to revisit my version specifiers :scream: 

But yes, I think this is much clearer:
```suggestion
numpy==1.19.1; python_version < "3.10"
numpy==1.21.4; python_version >= "3.10"
```

---

It does indeed work. But anyway, have already made the change.

---

Oh, this is [such a fail](https://www.python.org/dev/peps/pep-0496/#version-numbers):
- `python_version` is `platform.python_version()[:3]`
- only `python_full_version` includes the patch version

---

Not to mention that `[:3]` translates `3.10.1` to `3.1` :laughing: